In [1]:
import pandas as pd


In [2]:
imdb_data = pd.read_csv('IMDB-Movie-Data.csv')

In [3]:
imdb_data

Rank                    Title                     Genre  \
0       1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2               Prometheus  Adventure,Mystery,Sci-Fi   
2       3                    Split           Horror,Thriller   
3       4                     Sing   Animation,Comedy,Family   
4       5            Suicide Squad  Action,Adventure,Fantasy   
..    ...                      ...                       ...   
995   996     Secret in Their Eyes       Crime,Drama,Mystery   
996   997          Hostel: Part II                    Horror   
997   998   Step Up 2: The Streets       Drama,Music,Romance   
998   999             Search Party          Adventure,Comedy   
999  1000               Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
995  A tight-knit team of rising investigators, alo...             Billy Ray   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
998  A pair of friends embark on a mission to reuni...        Scot Armstrong   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
995  Chiwetel Ejiofor, Nicole Kidman, Julia Roberts...  2015   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
998  Adam Pally, T.J. Miller, Thomas Middleditch,Sh...  2014   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
1                  124     7.0  485820              126.46       65.0  
2                  117     7.3  157606              138.12       62.0  
3                  108     7.2   60545              270.32       59.0  
4                  123     6.2  393727              325.02       40.0  
..                 ...     ...     ...                 ...        ...  
995                111     6.2   27585                 NaN       45.0  
996                 94     5.5   73152               17.54       46.0  
997                 98     6.2   70699               58.01       50.0  
998                 93     5.6    4881                 NaN       22.0  
999                 87     5.3   12435               19.64       11.0  

[1000 rows x 12 columns]

In [5]:
type(imdb_data) # confirm the data type

pandas.core.frame.DataFrame

# Question 1: How many films were produced in 2015?

In [7]:
imdb_data.dropna(how='any', inplace=True) # Clean the dataset by removing null values

In [8]:
imdb_data

Rank                     Title                     Genre  \
0       1   Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2                Prometheus  Adventure,Mystery,Sci-Fi   
2       3                     Split           Horror,Thriller   
3       4                      Sing   Animation,Comedy,Family   
4       5             Suicide Squad  Action,Adventure,Fantasy   
..    ...                       ...                       ...   
993   994  Resident Evil: Afterlife   Action,Adventure,Horror   
994   995                 Project X                    Comedy   
996   997           Hostel: Part II                    Horror   
997   998    Step Up 2: The Streets       Drama,Music,Romance   
999  1000                Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
993  While still out to destroy the evil Umbrella C...    Paul W.S. Anderson   
994  3 high school seniors throw a birthday party t...       Nima Nourizadeh   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
993  Milla Jovovich, Ali Larter, Wentworth Miller,K...  2010   
994  Thomas Mann, Oliver Cooper, Jonathan Daniel Br...  2012   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
1                  124     7.0  485820              126.46       65.0  
2                  117     7.3  157606              138.12       62.0  
3                  108     7.2   60545              270.32       59.0  
4                  123     6.2  393727              325.02       40.0  
..                 ...     ...     ...                 ...        ...  
993                 97     5.9  140900               60.13       37.0  
994                 88     6.7  164088               54.72       48.0  
996                 94     5.5   73152               17.54       46.0  
997                 98     6.2   70699               58.01       50.0  
999                 87     5.3   12435               19.64       11.0  

[838 rows x 12 columns]

In [21]:
imdb_data[imdb_data["Year"] == 2015]["Year"].count() # Get all movies produced in 2015 and value count

50     2015
63     2015
67     2015
84     2015
85     2015
       ... 
923    2015
927    2015
950    2015
961    2015
984    2015
Name: Year, Length: 109, dtype: int64

# The total number of movies produced in 2015 is 109

# How many action movies were produced in 2016?

In [70]:
new = imdb_data[(imdb_data["Year"] == 2016) & (imdb_data["Genre"].str.contains("Action"))]

In [71]:
new[["Title","Genre", "Year"]]

Title                       Genre  Year
4         Suicide Squad    Action,Adventure,Fantasy  2016
5        The Great Wall    Action,Adventure,Fantasy  2016
8    The Lost City of Z  Action,Adventure,Biography  2016
12            Rogue One     Action,Adventure,Sci-Fi  2016
14             Colossal         Action,Comedy,Drama  2016
..                  ...                         ...   ...
615        Bastille Day          Action,Crime,Drama  2016
679  The Age of Shadows       Action,Drama,Thriller  2016
741      Hands of Stone      Action,Biography,Drama  2016
787           Max Steel     Action,Adventure,Family  2016
880         Shin Gojira      Action,Adventure,Drama  2016

[62 rows x 3 columns]

## Which Director's movies made the most revenue in the dataset?

In [61]:
revenue_by_director = imdb_data.groupby('Director')[['Revenue (Millions)']].sum()

In [66]:
revenue_by_director.sort_values(by="Revenue (Millions)", ascending=False)

Revenue (Millions)
Director                             
J.J. Abrams                   1683.45
David Yates                   1630.51
Christopher Nolan             1515.09
Michael Bay                   1421.32
Francis Lawrence              1299.81
...                               ...
Gus Van Sant                     0.02
Robin Swicord                    0.01
So Yong Kim                      0.01
Patricia Rozema                  0.01
Andy Goddard                     0.00

[524 rows x 1 columns]

## J.J Abrams Films made the most revenue